// coverage-badges-cli/lib/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.exampleHelp = exports.cliHelp = void 0;
const minimist_1 = __importDefault(require("minimist"));
const create_1 = require("./create");
function run() {
    const argvs = (0, minimist_1.default)(process.argv.slice(2), {
        alias: {
            help: 'h',
            source: 's',
            output: 'o',
        },
        default: {
            style: 'classic',
            source: 'coverage/coverage-summary.json',
            output: 'coverage/badges.svg',
            jsonPath: 'total.statements.pct',
        },
    });
    if (argvs.h || argvs.help) {
        cliHelp();
        exampleHelp();
        return;
    }
    const { version } = require('../package.json');
    if (argvs.v || argvs.version) {
        console.log(`\n coverage-badges-cli v${version}\n`);
        return;
    }
    (0, create_1.create)(argvs);
}
exports.default = run;
function cliHelp() {
    console.log('\n  Usage: coverage-badges [options] [--help|h]');
    console.log('\n  Options:\n');
    console.log('    --version, -v ', 'Show version number');
    console.log('    --help, -h    ', 'Displays help information.');
    console.log('    --output, -o  ', 'Output directory.');
    console.log('    --source, -s  ', 'The path of the target file "coverage-summary.json".');
    console.log('    --style       ', 'Badges style: flat, flat-square.');
}
exports.cliHelp = cliHelp;
function exampleHelp() {
    console.log('\n  Example:\n');
    console.log('    \x1b[35mnpm\x1b[0m coverage-badges-cli \x1b[33m--output\x1b[0m coverage/badges.svg');
    console.log('    \x1b[35mnpm\x1b[0m coverage-badges-cli \x1b[33m--style\x1b[0m plastic');
    console.log('    \x1b[35mnpm\x1b[0m coverage-badges-cli \x1b[33m--source\x1b[0m coverage/coverage-summary.json');
    console.log('\n');
}
exports.exampleHelp = exampleHelp;
